### ISSUE TITLE
iOS - duplicate symbols for architecture arm64

### ISSUE THREAD

**Issue by ngdminh93** [Pushwoosh/pushwoosh-flutter#28]
Hello, I'm trying to implement pushwoosh library to my app using the docs below, but it doesn't work 
https://docs.pushwoosh.com/platform-docs/pushwoosh-sdk/cross-platform-frameworks/flutter

**step to procedure**
- flutter create my_app
- add 'pushwoosh: ^1.20.1' to pubspec.yaml file
- add pod 'Pushwoosh' to ios/Podfile
- flutter launch

**error log**
```
Launching lib/main.dart on iPhone in debug mode...
 lib/main.dart
Signing iOS app for device deployment using developer identity: 
Xcode build done.                                            6.2s
Failed to build iOS app
Error output from Xcode build:
↳
    2021-01-25 11:28:49.002 xcodebuild[31897:607242] CFURLRequestSetHTTPCookieStorageAcceptPolicy_block_invoke: no longer implemented and should not be called
    ** BUILD FAILED **
Xcode's output:
↳
    In file included from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/PushwooshPlugin.m:2:
    In file included from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/Pushwoosh.h:7:
    /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/PushNotificationManager.h:245:44: warning: 'UNUserNotificationCenterDelegate' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    @property (nonatomic, strong, readonly) id<UNUserNotificationCenterDelegate> notificationCenterDelegate;
                                               ^
    In module 'UserNotifications' imported from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/PushNotificationManager.h:12:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.3.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNUserNotificationCenter.h:89:11: note: 'UNUserNotificationCenterDelegate' has been marked as being introduced in iOS 10.0 here, but the deployment target is iOS 8.0.0
    @protocol UNUserNotificationCenterDelegate <NSObject>
              ^
    In file included from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/PushwooshPlugin.m:2:
    In file included from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/Pushwoosh.h:7:
    /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/PushNotificationManager.h:202:12: note: annotate 'PushNotificationManager' with an availability attribute to silence this warning
    @interface PushNotificationManager : NSObject {
               ^
    In file included from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/PushwooshPlugin.m:2:
    In file included from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/Pushwoosh.h:18:
    /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/PWNotificationExtensionManager.h:29:36: warning: 'UNNotificationRequest' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - (void)handleNotificationRequest:(UNNotificationRequest *)request contentHandler:(void (^)(UNNotificationContent *))contentHandler;
                                       ^
    In module 'UserNotifications' imported from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/PushNotificationManager.h:12:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.3.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNNotificationRequest.h:16:12: note: 'UNNotificationRequest' has been marked as being introduced in iOS 10.0 here, but the deployment target is iOS 8.0.0
    @interface UNNotificationRequest : NSObject <NSCopying, NSSecureCoding>
               ^
    In file included from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/PushwooshPlugin.m:2:
    In file included from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/Pushwoosh.h:18:
    /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/PWNotificationExtensionManager.h:29:1: note: annotate 'handleNotificationRequest:contentHandler:' with an availability attribute to silence this warning
    - (void)handleNotificationRequest:(UNNotificationRequest *)request contentHandler:(void (^)(UNNotificationContent *))contentHandler;
    ^
                                                                                                                                        API_AVAILABLE(ios(10.0))
    /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/PWNotificationExtensionManager.h:29:93: warning: 'UNNotificationContent' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - (void)handleNotificationRequest:(UNNotificationRequest *)request contentHandler:(void (^)(UNNotificationContent *))contentHandler;
                                                                                                ^
    In module 'UserNotifications' imported from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/PushNotificationManager.h:12:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.3.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UNNotificationContent.h:16:12: note: 'UNNotificationContent' has been marked as being introduced in iOS 10.0 here, but the deployment target is iOS 8.0.0
    @interface UNNotificationContent : NSObject <NSCopying, NSMutableCopying, NSSecureCoding>
               ^
    In file included from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/PushwooshPlugin.m:2:
    In file included from /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/Pushwoosh.h:18:
    /Users/abc/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/pushwoosh-1.20.1/ios/Classes/../Library/PWNotificationExtensionManager.h:29:1: note: annotate 'handleNotificationRequest:contentHandler:' with an availability attribute to silence this warning
    - (void)handleNotificationRequest:(UNNotificationRequest *)request contentHandler:(void (^)(UNNotificationContent *))contentHandler;
    ^
                                                                                                                                        API_AVAILABLE(ios(10.0))
    3 warnings generated.
    duplicate symbol '_OBJC_CLASS_$_PodsDummy_pushwoosh' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(pushwoosh-dummy.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(pushwoosh-dummy.o)
    duplicate symbol '_OBJC_METACLASS_$_PodsDummy_pushwoosh' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(pushwoosh-dummy.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(pushwoosh-dummy.o)
    duplicate symbol '_OBJC_IVAR_$_PushwooshPlugin._registerResult' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_IVAR_$_PushwooshPlugin._acceptHandler' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_IVAR_$_PushwooshPlugin._openHandler' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_CLASS_$_DeepLinkStreamHandler' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_METACLASS_$_DeepLinkStreamHandler' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_CLASS_$_PushwooshStreamHandler' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_METACLASS_$_PushwooshStreamHandler' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_IVAR_$_PushwooshPlugin._receiveHandler' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_IVAR_$_PushwooshStreamHandler._startPushNotification' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_CLASS_$_PushwooshPlugin' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_METACLASS_$_PushwooshPlugin' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_IVAR_$_DeepLinkStreamHandler._eventSink' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_IVAR_$_PushwooshStreamHandler._eventSink' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_IVAR_$_DeepLinkStreamHandler._cachedDeepLink' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    duplicate symbol '_OBJC_IVAR_$_PushwooshPlugin._cachedDeepLink' in:
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/Pushwoosh.framework/Pushwoosh(PushwooshPlugin.o)
        /Users/abc/Developer/work/flutter_test_pushwoosh/flutter_application_1/build/ios/Debug-iphoneos/pushwoosh/pushwoosh.framework/pushwoosh(PushwooshPlugin.o)
    ld: 17 duplicate symbols for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target 'pushwoosh' from project 'Pods')
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target 'Flutter' from project 'Pods')
Could not build the precompiled application for the device.

It appears that your application still contains the default signing identifier.
Try replacing 'com.example' with your signing id in Xcode:
  open ios/Runner.xcworkspace

Error launching application on iPhone.
Exited (sigterm) 
```

**Podfile**
```
# Uncomment this line to define a global platform for your project
platform :ios, '9.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  pod 'Pushwoosh'
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
```

**flutter doctor -v**
```
[✓] Flutter (Channel stable, 1.22.5, on macOS 11.1 20C69 darwin-x64, locale en-VN)
    • Flutter version 1.22.5 at /Users/abc/Developer/flutter
    • Framework revision 7891006299 (6 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/abc/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin not installed; this adds Flutter specific functionality.
    • Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1
```

**Comment by wfhm** [comment]
@ngdminh93,

> - **add pod 'Pushwoosh' to ios/Podfile**

This step is excessive. Just run `flutter build ios` and all required dependencies will be added automatically. Just tried it with a fresh project and it worked for me just fine.

**Comment by ngdminh93** [comment]
Thank you for updating your documents, but I have another issue with the code, I'd love to post it here:

In android, the below code seems to work fine, but not in iOS, it will through an exception
Pushwoosh.getInstance.setMultiNotificationMode(true);

![Screen Shot 2021-01-28 at 19 50 18](https://user-images.githubusercontent.com/43509518/106145909-a674bb00-61a8-11eb-9526-aa9ff2b39553.png)

**Comment by wfhm** [comment]
@ngdminh93,

`Pushwoosh.getInstance.setMultiNotificationMode(true)` method is created only for Android, iOS pushes do not overwrite existing ones by default. We will fix the issue with the new plugin version, meanwhile, you can perform a platform check in your code with `Platform.isAndroid` and run this method only for Android:

https://api.flutter.dev/flutter/dart-io/Platform/isAndroid.html

**Comment by ngdminh93** [comment]
Thank you @wfhm, I facing another issue. I have set badge +1 while sending the message, in iOS when I click the notification or open the app, the badge not reset to 0. Any advice?

**Comment by wfhm** [comment]
@ngdminh93,

I am sorry for the inconvenience here - it looks like we did not add the method to work with badges when creating bindings for our native SDKs. I have added the task to add it with the next release. Meanwhile, for iOS, as a workaround, it can be done with any badge plugin.

Also, may I ask you to create separate tickets for different issues?

**Comment by ngdminh93** [comment]
Oh sorry, I'm new to github. next time I will create separate tickets. For all the above issues, hope to hear from you soon

**Comment by wfhm** [comment]
@ngdminh93 fixed with the 2.0.0 release.
